src/Umbraco.AuthorizedServices/Services/Implement/AuthorizedServiceCaller.cs

```diff
@@ -80,7 +81,13 @@ public async Task<string> SendRequestRawAsync<TRequest>(string serviceAlias, str
         HttpRequestMessage requestMessage;
         if (serviceDetail.AuthenticationMethod == AuthenticationMethod.ApiKey)
         {
-            requestMessage = _authorizedRequestBuilder.CreateRequestMessageWithApiKey(serviceDetail, path, httpMethod, requestContent);
+            string? key = KeyStorage.GetKey(serviceAlias);
```

Comment:
Not quite sure this flow is quite right, could consider:

- Check for a configured  API key first.  If we have one, we can use it without having to go to the database to check for a key.
- Only if one isn't configured should we look for a stored key.
- With the token flow, we throw an exception if we haven't got a token.  Seems we should do the same here, if we don't have a configured or a stored key.

Reply:
Updated